### DISCUSSION TITLE
GitHub API: List repositories endpoint does not filter by `type=sources`

### DISCUSSION THREAD

**Discussion by ferrata** [community/community#39719] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

Hey there!

It looks like I've found an issue with the GitHub repositories API. Repositories endpoint `/repos` does not filter by `type=sources`.

Here is what I see in [the documentation](https://docs.github.com/en/rest/repos/repos#list-organization-repositories):
> `type` string
> Specifies the types of repositories you want returned. ...
> ...
> Can be one of: `all`, `public`, `private`, `forks`, `sources`, `member`, `internal`

However, when I call an API at: `api.github.com/users/<user>/repos?type=sources` it also returns forks, which I think it should not.

Also, the result is identical to the one that I would get with `?type=public`

Is this a known issue?
Thanks!

**Comment by LangLangBart** [comment, marked as answer]
The link in your bug report points to [List organization repositories](https://docs.github.com/en/rest/repos/repos#list-organization-repositories) with the syntax: `/orgs/{org}/repos`

The link you provided has the syntax: `api.github.com/users/<user>/repos...` which corresponds to [List repositories for a user](https://docs.github.com/en/rest/repos/repos#list-repositories-for-a-user).

> type string
> Limit results to repositories of the specified type.

> Default: owner

> Can be one of: all, owner, member

**Reply by ferrata** [reply]
Yes, you are right! Sorry, my mistake 😢

**Reply by FernandoZX** [reply]
And how to return the result with type `private` because i followed the documentation and returns an empty array.

**Reply by ilia-beliaev-miro** [reply]
Hey @FernandoZX did you find a way to do it?
